src/Infrastructure/Event/Workflow/WorkflowEventListener.php

```diff
@@ -76,12 +76,13 @@ public function setup(ConsoleEvent $event): void
      */
     public function handle(Event $event): void
     {
-        match ($event::class) {
-            GuardEvent::class => $this->guard($event),
-            LeaveEvent::class => $this->event($event, static::WORKFLOW_BEFORE),
-            EnteredEvent::class => $this->event($event, static::WORKFLOW_AFTER),
-            default => false,
-        };
+        if ($event instanceof GuardEvent) {
```

Comment:
`$event instanceof GuardEvent` not equals `$event::class`. `get_class` more suitable here. And maybe move it to switch-case?

Reply:
Ok, but: all classes to the right of the `instanceof` have the same parent and are final, so instanceof will work the same as get_class, and switch-case is 1.5 times more lines of code.

Reply:
switch()

Reply:
well, it is two opinions versus one, changed to switch-case